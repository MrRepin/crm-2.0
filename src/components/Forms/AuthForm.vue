<template>
    <v-col md="4">
        <v-card class="auth-form">
            <ColorIcon :icon="data.icon" />
            <p class="text-left form-title font-weight-light">{{data.title}}</p>
            <v-form
                ref="form"
                @submit.prevent="submitForm"
            >
                <v-text-field
                    v-model="email"
                    label="E-mail"
                    :error-messages="emailErrors"
                    required
                    @input="$v.email.$touch()"
                    @blur="$v.email.$touch()"
                ></v-text-field>

                <v-text-field
                    v-model="password"
                    label="Password"
                    :error-messages="passErrors"
                    required
                    @input="$v.password.$touch()"
                    @blur="$v.password.$touch()"
                ></v-text-field>

                <v-btn
                    color="success"
                    class="mr-4"
                    type="submit"
                >
                    Send
                </v-btn>
            </v-form>
        </v-card>
        <v-btn class="change-forms" text color="primary" @click="$emit('change-form', !form)">{{data.button}}</v-btn>
    </v-col>
</template>

<script>

import {mapActions} from 'vuex'
import {email, required, minLength} from 'vuelidate/lib/validators'
import ColorIcon from '@/components/Icons/ColorIcon'
import formMixin from '@/mixins/FormMixin'

export default {
    name: 'AuthForm',
    mixins: [formMixin],
    props: {
        data: Object,
        form: Boolean,
    },
    data: () => ({
        email: '',
        password: ''
    }),
    validations: {
        email: {
            email,
            required,
        },
        password: {
            minLength: minLength(6),
            required
        },
    },
    methods: {
        ...mapActions(['signIn']),
        async submitForm() {
            if(this.$v.$invalid) {
                this.$v.$touch()
                return
            }
            const signInData = {
                email: this.email,
                password: this.password
            }
            try {
                await this.signIn(signInData)
                this.$router.push('/')
            } catch(e) {
                console.log(e)
            }
        }
    },
    components: {
        ColorIcon
    }
}
</script>

<style lang="sass" scoped>
    .auth-form
        position: relative

    .form-title
        font-size: 25px
        margin-left: 80px

    .change-forms
        margin: 20px auto 0
        display: block
</style>