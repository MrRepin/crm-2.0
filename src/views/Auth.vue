<template>
    <div class="auth">
        <v-container fluid>
            <v-row>
                <v-col cols="12">
                    <v-row align="center" justify="center">
                        <RegisterForm
                            v-if="form"
                            :form="form"
                            :data="regForm"
                            @change-form="changeForm"
                        />
                        <AuthForm
                            v-else
                            :form="form"
                            :data="authForm"
                            @change-form="changeForm"
                        />
                    </v-row>
                </v-col>
            </v-row>
        </v-container>
    </div>
</template>

<script>
import AuthForm from '@/components/Forms/AuthForm'
import RegisterForm from '@/components/Forms/RegisterForm'
export default {
    name: 'Auth',
    data: () => ({
        form: true,
        regForm: {
            title: 'Registration',
            icon: 'mdi-account-plus',
            button: 'Login'
        },
        authForm: {
            title: 'Login',
            icon: 'mdi-login-variant',
            button: 'Registration'
        },
    }),
    methods: {
        changeForm() {
            this.form = !this.form
        }
    },
    components: {
      AuthForm,
      RegisterForm
    },
}
</script>

<style lang="sass">
    .auth
        height: 100vh
        .container,
        .row,
        .col-12
            height: 100%

    .auth-form
        padding: 20px
</style>