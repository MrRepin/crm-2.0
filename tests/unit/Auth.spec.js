import Vue from 'vue'
import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Vuelidate from 'vuelidate'

import Auth from '@/views/Auth'
import AuthForm from '@/components/Forms/AuthForm'

Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(Vuelidate)

describe('Auth', () => {
  const wrapper = mount(Auth, {
    stubs: {
      AuthForm
    },
    localVue
  })
  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('is render correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('contains AuthForm', () => {
    expect(wrapper.findComponent(AuthForm).element).toBeTruthy()
  })
})
