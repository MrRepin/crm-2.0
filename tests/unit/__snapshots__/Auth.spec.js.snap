// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth is render correctly 1`] = `
<div
  class="auth"
>
  <div
    class="container container--fluid"
  >
    <div
      class="row"
    >
      <div
        class="col col-12"
      >
        <div
          class="row align-center justify-center"
        >
          <div
            class="col-md-4 col"
          >
            <div
              class="auth-form v-card v-sheet theme--light"
            >
              <div
                class="icon color-bg-icon elevation-4 green"
              >
                <div
                  class="v-list-item__icon"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-account-plus theme--light"
                  />
                </div>
              </div>
               
              <p
                class="text-left form-title font-weight-light"
              >
                Registration
              </p>
               
              <form
                class="v-form"
                novalidate="novalidate"
              >
                <div
                  class="v-input theme--light v-text-field"
                >
                  <div
                    class="v-input__control"
                  >
                    <div
                      class="v-input__slot"
                    >
                      <div
                        class="v-text-field__slot"
                      >
                        <label
                          class="v-label theme--light"
                          for="input-13"
                          style="right: 0px; position: absolute;"
                        >
                          Name
                        </label>
                        <input
                          id="input-13"
                          required="required"
                          type="text"
                        />
                      </div>
                    </div>
                    <div
                      class="v-text-field__details"
                    >
                      <div
                        class="v-messages theme--light"
                      >
                        <transition-group-stub
                          class="v-messages__wrapper"
                          name="message-transition"
                          tag="div"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="v-input theme--light v-text-field"
                >
                  <div
                    class="v-input__control"
                  >
                    <div
                      class="v-input__slot"
                    >
                      <div
                        class="v-text-field__slot"
                      >
                        <label
                          class="v-label theme--light"
                          for="input-16"
                          style="right: 0px; position: absolute;"
                        >
                          E-mail
                        </label>
                        <input
                          id="input-16"
                          required="required"
                          type="text"
                        />
                      </div>
                    </div>
                    <div
                      class="v-text-field__details"
                    >
                      <div
                        class="v-messages theme--light"
                      >
                        <transition-group-stub
                          class="v-messages__wrapper"
                          name="message-transition"
                          tag="div"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="v-input theme--light v-text-field"
                >
                  <div
                    class="v-input__control"
                  >
                    <div
                      class="v-input__slot"
                    >
                      <div
                        class="v-text-field__slot"
                      >
                        <label
                          class="v-label theme--light"
                          for="input-19"
                          style="right: 0px; position: absolute;"
                        >
                          Password
                        </label>
                        <input
                          id="input-19"
                          required="required"
                          type="text"
                        />
                      </div>
                    </div>
                    <div
                      class="v-text-field__details"
                    >
                      <div
                        class="v-messages theme--light"
                      >
                        <transition-group-stub
                          class="v-messages__wrapper"
                          name="message-transition"
                          tag="div"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="v-input theme--light v-input--selection-controls v-input--checkbox"
                >
                  <div
                    class="v-input__control"
                  >
                    <div
                      class="v-input__slot"
                    >
                      <div
                        class="v-input--selection-controls__input"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon notranslate material-icons theme--light"
                        >
                          $checkboxOff
                        </i>
                        <input
                          aria-checked="false"
                          id="input-22"
                          required="required"
                          role="checkbox"
                          type="checkbox"
                          value=""
                        />
                        <div
                          class="v-input--selection-controls__ripple"
                        />
                      </div>
                      <label
                        class="v-label theme--light"
                        for="input-22"
                        style="left: 0px; position: relative;"
                      >
                        Do you agree?
                      </label>
                    </div>
                    <div
                      class="v-messages theme--light"
                    >
                      <transition-group-stub
                        class="v-messages__wrapper"
                        name="message-transition"
                        tag="div"
                      />
                    </div>
                  </div>
                </div>
                 
                <button
                  class="mr-4 v-btn v-btn--contained theme--light v-size--default success"
                  type="submit"
                >
                  <span
                    class="v-btn__content"
                  >
                    
                Send
            
                  </span>
                </button>
              </form>
            </div>
             
            <button
              class="change-forms v-btn v-btn--flat v-btn--text theme--light v-size--default primary--text"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                Login
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
